src/cwrapper.cpp

```diff
@@ -77,6 +77,21 @@ void integer_set_str(basic s, char* c)
     *RCP_cast(s) = CSymPy::integer(mpz_class(c, 10));
 }
 
+signed long integer_get_si(const basic s)
+{
```

Comment:
There should be `CSYMPY_ASSERT(is_a<Integer>(*RCP_const_cast(s)))` here. Otherwise this can segfault even in Debug mode, if `s` is not an Integer. This should also be fixed at other places in `cwrapper.cpp`.